engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -1094,6 +1069,9 @@ class JoinPartitionKernel : public kernel {
 	std::vector<cudf::size_type> left_column_indices, right_column_indices;
 	std::vector<cudf::data_type> join_column_common_types;
 	bool normalize_left, normalize_right;
+	ral::distribution::MessageManager message_manager_left;
```

Comment:
it seems pretty silly to have  message_manager per kernel. Instead just make a few utility functions and leave the logic the same as i twas before.

Reply:
One of the key ideas behind the message manager is that it maintains the state necessary for proper sending and receiving of messages such as the node_count

Reply:
The kernel should manage this as it did before.  Basically you are performring broadcast / gather. I would be ok with a broadcast function and one for gather that was a bit more clear but this  just seems more confusing